src/common/token.ts

```diff
@@ -14,11 +14,15 @@
  * You should have received a copy of the GNU Affero General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-import Arweave from 'arweave';
-import * as Transaction from 'arweave/node/lib/transaction.js';
+import Arweave from '@irys/arweave';
```

Comment:
can we make a PR/issue request to arweave-js with the solution irys created?

Reply:
I'm not confident I can describe the problem and solution well enough. It seems like its the ESM web context that becomes wrongly the NodeArweave in the `arweave-js` version. While the irys implementation handles it well

FWIW its an MIT solution, and I've locked the version on it. We can easily switch back to the other dependency if/when it becomes compatible with ESM in the browser environment

Reply:
honestly I have the urge (and half started) to just write up arweave Tx creation/handling here instead of this but I mainly don't want to reinvent the wheel on node/web signature details